SEDaily-IOS/GeneralCollectionViewController.swift

```diff
@@ -18,281 +16,144 @@ class GeneralCollectionViewController: UICollectionViewController {
-        
+
         // Register cell classes
         self.collectionView!.register(PodcastCell.self, forCellWithReuseIdentifier: reuseIdentifier)
         
-        self.collectionView?.backgroundColor = UIColor(hex: 0xfafafa)
-        self.collectionView?.showsHorizontalScrollIndicator = false
-        self.collectionView?.showsVerticalScrollIndicator = false
-        
-        let layout = KoalaTeaFlowLayout(cellWidth: 158,
-                                       cellHeight: 250,
-                                       topBottomMargin: 12,
-                                       leftRightMargin: 20,
-                                       cellSpacing: 8)
+        let layout = KoalaTeaFlowLayout(cellWidth: UIView.getValueScaledByScreenWidthFor(baseValue: 158),
+                                        cellHeight: UIView.getValueScaledByScreenHeightFor(baseValue: 250),
+                                        topBottomMargin: UIView.getValueScaledByScreenHeightFor(baseValue: 12),
+                                        leftRightMargin: UIView.getValueScaledByScreenWidthFor(baseValue: 20),
+                                        cellSpacing: UIView.getValueScaledByScreenWidthFor(baseValue: 8))
         self.collectionView?.collectionViewLayout = layout
+        self.collectionView?.backgroundColor = .white
+        
+        // Load initial data
+        self.getData(lastIdentifier: "", nextPage: 0)
         
         // User Login observer
         NotificationCenter.default.addObserver(self, selector: #selector(self.loginObserver), name: .loginChanged, object: nil)
         
-        
         self.collectionView?.addSubview(skeletonCollectionView)
     }
     
-    override func didReceiveMemoryWarning() {
-        super.didReceiveMemoryWarning()
-        // Dispose of any resources that can be recreated.
-    }
-    
     @objc func loginObserver() {
-        loadData(lastItemDate: "")
-    }
-    
-    // MARK: - Data stuff
-    
-    func getData(page: Int = 0, lastItemDate: String = "") {
-        guard self.loading == false else { return }
-        
-        lastLoadedPage = page
-        
-        self.loading = true
-        loadData(lastItemDate: lastItemDate)
-    }
-    
-    func checkPage(indexPath: IndexPath, item: PodcastModel) {
-        let nextPage: Int = Int(indexPath.item / pageSize) + 1
-        let preloadIndex = nextPage * pageSize - preloadMargin
-        
-        if (indexPath.item >= preloadIndex && lastLoadedPage < nextPage) || indexPath == collectionView?.indexPathForLastItem! {
-            if let lastDate = item.uploadDate {
-                //@TODO: This is left open for paging for recommended and top posts
-                // (I think top posts could be paged)
-                guard type != API.Types.recommended || type != API.Types.top else { return }
-                getData(page: nextPage, lastItemDate: lastDate)
+        if self.type == .recommended {
```

Comment:
This looks like a specific use case. Would it be better to do this in CustomTabViewController, which is one level up?

Reply:
I'm not sure how this would look in CustomTabViewController but I'm open.

The better fix would probably be in the design of the app. We probably shouldn't show the top podcasts in the Recommended tab. We may just have a view that prompts to login to start receiving recommended podcasts. Not 100% on this of course.

Reply:
I was thinking of setting up the sign in listener in CustomTabViewController. When you detect a log in, you trigger a clear on the recommended GeneralCollectionViewController. GeneralCollectionViewController should have a public function to clear its model view and reload data. This separates out use case and functionality.

Reply:
That could definitely work. I think we should work on fixing the memory issue and there is also a possibility we may need to refresh all models to retrieve a logged in users upvoted and downvoted podcasts. These issues will all be closely coupled and our approach will fix all of these at once.